app/src/main/java/org/fossasia/openevent/app/core/event/list/pager/ListPageFragment.java

```diff
@@ -0,0 +1,167 @@
+        context = getActivity();
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_list_page, container, false);
+
+        position = getArguments().getInt("position");
+        eventsViewModel = ViewModelProviders.of(getActivity(), viewModelFactory).get(EventsViewModel.class);
+
+        return binding.getRoot();
+    }
+
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.menu_events, menu);
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        setupRecyclerView();
+
+        initialized = true;
+
+        eventsViewModel.getEvents(position).observe(this, (events) -> {
+            eventListAdapter.updateList(events);
+
+            if (events == null || events.isEmpty())
+                this.showEmptyView(true);
+            else
+                this.showEmptyView(false);
+        });
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        eventListAdapter.unregisterAdapterDataObserver(adapterDataObserver);
+    }
+
+    private void setupRecyclerView() {
+        if (!initialized) {
+            eventListAdapter = new ListPageAdapter(eventsViewModel.getEvents(position).getValue(), bus, this);
+
+            recyclerView = binding.eventRecyclerView;
+            recyclerView.setLayoutManager(new LinearLayoutManager(context));
+            recyclerView.setAdapter(eventListAdapter);
+            recyclerView.addItemDecoration(new DividerItemDecoration(context, DividerItemDecoration.VERTICAL));
+            recyclerView.setItemAnimator(new DefaultItemAnimator());
+            StickyRecyclerHeadersDecoration decoration = new StickyRecyclerHeadersDecoration(eventListAdapter);
+            recyclerView.addItemDecoration(decoration);
+
+            adapterDataObserver = new RecyclerView.AdapterDataObserver() {
+                @Override
+                public void onChanged() {
+                    decoration.invalidateHeaders();
+                }
+            };
+        }
+        eventListAdapter.registerAdapterDataObserver(adapterDataObserver);
+    }
+
+    @Override
+    public void showResults(List<Event> events) {
+        eventListAdapter.updateList(events);
+    }
+
+    @Override
+    public void showEmptyView(boolean show) {
+        ViewUtils.showView(binding.emptyView, show);
+    }
+
+    @Override
+    public void openSalesSummary(Long eventId) {
+        DialogFragment fragment = SalesSummaryFragment.newInstance(eventId);
+        fragment.show(getFragmentManager(), "summary");
+    }
+
+    @Override
+    public void closeSalesSummary() {
+        DialogFragment fragment = ((DialogFragment) getFragmentManager().findFragmentByTag("summary"));
```

Comment:
Use string variable for "summary".

Reply:
It's a tag and not a string that a user will see. No string resource is needed here.